command_line/powder_calibrate.py

```diff
@@ -0,0 +1,467 @@
+"""
+Calibrate geometry using powder tools.
+
+    `pyFAI` is a well established X-ray powder diffraction tool.
+https://doi.org/10.1107/S1600576715004306
+
+    `circle-fit` is a nifty module that fits an ellipse to a set of points,
+even if the points are covering a section as small as a quarter of the conic.
+https://doi.org/10.1016/j.csda.2010.12.012
+
+NOTE: The algorithms assumes that find_spots worked well at separating
+signal from noise especially close to the beam, assumption which might be hard
+to achieve for electron data.
+"""
+
+from sys import exit
+
+import numpy as np
+from matplotlib import pyplot as plt
+from matplotlib.pyplot import subplots
+from scipy.signal import find_peaks
+
+import dxtbx.flumpy as flumpy
+from iotbx.phil import parse
+
+
+def module_exists(module_name):
+    try:
+        __import__(module_name)
+    except ModuleNotFoundError as err:
+        if module_name == "circle_fit":
+            module_name = "circle-fit"
+        print(
+            err,
+            f"""
+            This script requires the {module_name} library. Try:
+
+                conda install -c conda-forge {module_name}
```

Comment:
I gave up with this after some minutes and used `mamba` instead. Perhaps we should push the user to `conda install` `mamba` first and then `mamba install -c conda-forge pyFAI`?

Reply:
My bad, `dials.powder_calibrate` was not meant to be in this branch at all.